app/views/new_filters/_location_provider_filter.html.erb

```diff
@@ -0,0 +1,30 @@
+<div class="app-filter__group" data-qa="filters__area_and_provider">
+  <legend class="govuk-fieldset__legend govuk-fieldset__legend--s">
+    <h3 class="govuk-fieldset__heading">
+      <% if @filters_view.provider_query? %>
+        Provider
+      <% else %>
+        Location
+      <% end %>
+    </h3>
+  </legend>
+  <p class="govuk-body-s govuk-!-margin-bottom-2" data-qa="area_or_provider_name">
+    <% if @filters_view.across_england_query? %>
+      Across England
+    <% elsif @filters_view.location_query? %>
+      <%= request.parameters['lq'] %>
+    <% elsif @filters_view.provider_query? %>
+      <%= request.parameters['query'] %>
+    <% end %>
+  </p>
+
+  <% if @filters_view.provider_query? %>
+    <%= govuk_link_to "Change provider or choose a location",
+      @results_view.filter_path_with_unescaped_commas(location_path),
+      data:  { qa: "filters__area_and_provider_link" } %>
+  <% else %>
+    <%= govuk_link_to "Change location or choose a provider",
+      @results_view.filter_path_with_unescaped_commas(location_path),
+      data:  { qa: "filters__area_and_provider_link" } %>
+  <% end %>
```

Comment:
Just had a play with the filters. This is working nicely when you first hit the results page, but if you use the filters it doesn't retain the selected providers.

Reply:
Yep I see what the issue is here ... we need to pass in the location query params as hidden fields. Have pushed a fix

Reply:
Do we also need to tdo his for the :query param to retain the selected provider?